snapcraft/_store.py

```diff
@@ -281,3 +283,92 @@ def download(snap_name, channel, download_path, arch):
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def _get_snap_data(name, store):
+    try:
+        details = store.cpi.get_package(name, 'stable', None)
+    except storeapi.errors.SnapNotFoundError:
+        # The original error appears was not good:
+        # The "snapname" for None was not found in stable.
+        raise RuntimeError(
+            'Snap {name} cannot be found in the stable channel'.format(
+                name=name))
+    return details
+
+
+def gated(snap_name):
+    """Print list of snaps gated by snap_name."""
+    store = storeapi.StoreClient()
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = _get_snap_data(snap_name, store)
+    snap_id = snap_data['snap_id']
+    validations = store.get_validations(snap_id)
+
+    table_data = []
+    for v in validations:
+        table_data.append([v['approved-snap-name'],
+                           v['approved-snap-revision']])
+    tabulated = tabulate(table_data, headers=['Name', 'Approved'],
+                         tablefmt="plain")
+    print(tabulated)
+
+
+def validate(snap_name, validations, revoke=False, key=None):
+    """Generate, sign and upload validation assertions."""
+
+    # Check validations format
+    validation_re = re.compile('^[^=]+=[0-9]+$')
+    invalids = [v for v in validations if not validation_re.match(v)]
+    if invalids:
+        for v in invalids:
+            logger.error('Invalid validation request "{}", format must be'
+                         ' name=revision'.format(v))
+        raise RuntimeError()
+
+    store = storeapi.StoreClient()
+
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = _get_snap_data(snap_name, store)
+    release = str(snap_data['release'][0])
+    snap_id = snap_data['snap_id']
+    developer_id = snap_data['developer_id']
```

Comment:
"developer_id" will be the account-id of the uploader of the latest revision available in stable channel, because of package-collaborating (sharing) it might not be the  "publisher", there is no "publisher_id" in CPI. More importantly, it's not the currently authenticated "account-id", so I think it will be less ambiguous (little more expensive) if you simply fetch account-info.account-id.